### ISSUE TITLE
Lost  wifi.creds text after multiple reboots

### ISSUE THREAD

**Issue by grey27** [anson-vandoren/esp8266-captive-portal#6]
code:
```
from captive_portal import CaptivePortal

portal = CaptivePortal()
contents = open("./wifi.creds", "rb").read().split(b",")
print()
print('VVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVV')
print(contents)
portal.start()
print(contents)
print('------------------------------------------')
```
logs：
```
# First restart
$l`rےol$ 

VVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVV
[b'MERCURY_6510', b'104104104104']
Trying to load WiFi credentials from ./wifi.creds
Trying to connect to SSID 'MERCURY_6510' with password 104104104104
#6 ets_task(4020f4d8, 28, 3fff9348, 10)
Connection in progress
Connection in progress
Connected to MERCURY_6510
Wrote credentials to ./wifi.creds
[b'MERCURY_6510', b'104104104104']
------------------------------------------
MicroPython v1.12 on 2019-12-20; ESP module with ESP8266
Type "help()" for more information.
>>> 

# Second restart
$l`
l$䌎2#
VVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVV
[b'']
Trying to load WiFi credentials from ./wifi.creds
Invalid credentials file: [b'']
Starting captive portal
Waiting for access point to turn on
#6 ets_task(4020f510, 29, 3fff8bc8, 10)
AP mode configured: ('192.168.4.1', '255.255.255.0', '192.168.4.1', '192.168.4.1')
HTTP Server listening on ('0.0.0.0', 80)
Configured HTTP server
```
I guess the text was not closed, so it was not saved
this problem can be solved by using ```with```
```
def write_creds(self, ssid, password):
        with open(self.CRED_FILE, "wb") as fp:
                fp.write(b",".join([ssid, password]))
        #open(self.CRED_FILE, "wb").write(b",".join([ssid, password]))
        print("Wrote credentials to {:s}".format(self.CRED_FILE))
```

**Comment by anson-vandoren** [comment]
Hi @grey27, thanks for submitting this. I don't have a lot of time to maintain this project right now, but happy to review & approve a pull request if you want to submit one.

**Comment by anson-vandoren** [comment]
Fixed in #7 

Thanks for the PR, @grey27

**Comment by anson-vandoren** [comment]
Fixed in #8 

Thanks, @grey27
